test/basic/optimalisationCheck.wast

```diff
@@ -0,0 +1,50 @@
+;; tests for Walrus where optimalisation shouldn't occure
+
+(module
+
+  ;; in the following tests there will be an EqualZero, then
+  ;; a JumpIfTrue/JumpIfFalse that should not be unified
+
+  (func (export "test1") (param $input i32) (result i32)
+    local.get $input
+    i32.eqz
+    i32.const 0
+    br_if 0
+  )
+
+  (func (export "test2") (param $input i32) (result i32)
+    local.get $input
+    i32.eqz
+    i32.const 0
+    if
+      nop
+    else
+      nop
+    end
+  )
+
+  (func (export "test3") (result i32)
+    i32.const 1
+    block (result i32)
+      i32.const 0
+      i32.const 1
+      br_if 0
+      ;; the latter part of the block
+      ;; is dead code
```

Comment:
Why this is dead code? It would be better if the argument of the br_if above would be a param and the code would be tested with and without jump.

Reply:
Its only purpose is to alter the second `br_if` (in case of a wrong bytecode unification). If this wrong unification occures then `JumpIfFalse` is changed to `JumpIfTrue` (but input value is still 0). This change results that the return value will be 1 (instead of 0).

In my humble opinion, there's no need for parameter.

What do you think?

Reply:
The test is better if it tests both with and without branch.

Reply:
Okay.